keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Constraint.kt

```diff
@@ -69,6 +70,34 @@ data class PipelineConstraint(
   val parameters: Map<String, Any?> = emptyMap()
 ) : StatefulConstraint("pipeline")
 
+// TODO: doesn't support sliding look-back, custom metric scope, or extended scope parameters (will default to asg)
+data class CanaryConstraint(
+  val serviceAccount: String = "anonymous",
+  val timeout: Duration = Duration.ofHours(24),
+  // TODO: resolve config name -> id via kayenta/canaryConfig?application=application
+  val canaryConfigId: String,
+  val beginAnalysisAfter: Duration = Duration.ofMinutes(10),
+  val canaryAnalysisInterval: Duration = Duration.ofMinutes(30),
+  val cleanupDelay: Duration = Duration.ZERO,
+  val lifetime: Duration,
+  // TODO: this default is only for testing; we need a configurable way to resolve constraint defaults -af
```

Comment:
I decided it was ok to leave as a todo for now since it can be set as needed in the constraint yml. I'll add a `ConfigurationProperties` class to redefine these defaults on a per-keel-install basis as a fast follow. The current hard-coded default (atlas-global.prod) will avoid the need to specify `metricsAccountName` for most (but not all) use-cases at Netflix. Every company that runs keel will likely want to define their own default based on their chosen telemetry platform and its configuration.